robusta_krr/core/integrations/prometheus/metrics_service/thanos_metrics_service.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional
+from kubernetes.client import ApiClient
+from requests.exceptions import ConnectionError, HTTPError
+
+from robusta_krr.core.models.config import Config
+from robusta_krr.utils.service_discovery import ServiceDiscovery
+
+from .prometheus_metrics_service import PrometheusMetricsService, PrometheusNotFound
+
+class ThanosMetricsDiscovery(ServiceDiscovery):
+    def find_metrics_url(self, *, api_client: Optional[ApiClient] = None) -> Optional[str]:
+        """
+        Finds the Thanos URL using selectors.
+        Args:
+            api_client (Optional[ApiClient]): A Kubernetes API client. Defaults to None.
+        Returns:
+            Optional[str]: The discovered Thanos URL, or None if not found.
+        """
+
+        return super().find_url(
+            selectors=[
+                "app.kubernetes.io/component=query,app.kubernetes.io/name=thanos",
+                "app.kubernetes.io/name=thanos-query",
+                "app=thanos-query",
+                "app=thanos-querier",
+            ]
+        )
+
+
+class ThanosMetricsNotFound(Exception):
```

Comment:
Should it be a subclass of `PrometheusNotFound`? Or maybe they should subclass some common parent aka `MetricsNotFound`? 

Because maybe we will want to catch all of them in one try block

Reply:
sounds good